src/components/Page/Page.tsx

```diff
@@ -94,36 +95,71 @@ export class Page extends React.PureComponent<ComposedProps, never> {
     );
   }
 
-  private hasHeaderContent() {
+  private get headerProps(): HeaderProps | undefined {
     const {
+      children,
+      fullWidth,
+      singleColumn,
+      polaris: {appBridge},
+      forceRenderTitle = false,
+      forceRenderActions = false,
       title,
       primaryAction,
       secondaryActions,
       actionGroups,
       breadcrumbs,
+      ...rest
     } = this.props;
 
-    return (
-      (title != null && title !== '') ||
-      primaryAction != null ||
-      (secondaryActions != null && secondaryActions.length > 0) ||
-      (actionGroups != null && actionGroups.length > 0) ||
-      (breadcrumbs != null && breadcrumbs.length > 0)
-    );
+    const hasHeaderTitle =
+      title != null && title !== '' && (appBridge ? forceRenderTitle : true);
```

Comment:
Quick note: whenever you're comparing the string `''`, always consider if you need to `trim()` the string first to avoid bugs using extra spaces at the end.

Reply:
Because this is the existing logic in Polaris, I am un-sure about changing the check @tmlayton double checking if this is the desired behaviour or can I introduce the trim check?